bus/pruning.go

```diff
@@ -57,7 +57,7 @@ func (b *Bus) pruneContractV1(ctx context.Context, rk types.PrivateKey, cm api.C
 	}
 
 	var pruneErr string
-	if err != nil {
+	if err != nil && !errors.Is(err, rhp2.ErrNoSectorsToPrune) {
```

Comment:
`nit` it would be mor readable if you'd handle `pruneErr` right after you check both `rhp2.ErrNoSectorsToPrune` and `context.Canceled`, so you'd end up with something like:

```go
var pruneErr string
err := b.rhp.Prune()
if err != nil && !checkErr1 && !checkErr2 {
    return
} else if err != nil {
    pruneErr = err.Error()
}
```